### ISSUE TITLE
Add a global toggle switch for enable/disable notifications

### ISSUE THREAD

**Issue by everestx** [shinken-monitoring/mod-webui#461]
Our organization is interested in contributing to webui2 by creating a global toggle switch for enable/disabling notifications.  This switch would a few purposes purposes:
 1. Allow action to enable/disable notifications
 1. Display the current status of enabled/disabled for a quick view.
 1. Allow people without access to our CLI access to scripts to enable/disable and view status of notifications

We currently accomplish this task via the webui using a CLI python script or a browser bookmark with the following URLs:
```
https://myshinkenhost/action/ENABLE_NOTIFICATIONS
https:///myshinkenhost/action/DISABLE_NOTIFICATIONS
```

It works great for periods of maintenance where we don't want to manage individual downtimes.  This should be managed only by shinken admins, based on the `isadmin` configuration in shinken contacts.

My question is where we should locate this global switch to accomodate other needs and the direction of the webui development?

Or would it make more sense to add a widget that anyone can use and put where they want?

**Comment by bittrance** [comment]
Generally speaking: +1.

I would argue that the better methodology would be to allow disabling notifications for a limited period. Any presentation of the feature should include the endtime or remainig time.

**Comment by everestx** [comment]
That's a good idea too.  I'm open to ideas on where to store that time information for notifications.  It's certainly very bad to have notifications disabled without people knowing about it.

**Comment by mohierf** [comment]
Many thanks for your proposal ...

 IMHO, there are several possible implementations:
 - in the sidebar menu, the *Configuration/Parameters* menu entry is present only when an administrator user is logged-in. This menu displays all the Shinken framework parameters and it may be a good starting point ... you may include a top toolbar on this page with action buttons.

 - a widget is probably the best idea for such a feature ... we could imagine to implement restrictions on the widgets that a user can include in his dashboard. Default is that everybody can include any widget ... but it would not be that hard to make some administrator widgets ... :+1: for this solution!

 Some information:
 - in the element view (*eltdetail* plugin) used to display information, you will find an implementation of a toglle button used to enable/disable hosts/services features
 - the *action* plugin manages the external commands sent to Shinken (ENABLE_NOTIFICATIONS)

 Anyway, do not hesitate to ask for information when you dive in the source code, because some features are quite messy to find :/ I started a huge refactoring to make some cleaning in the source code ...

**Comment by nchowning** [comment]
I started working on this yesterday. I've got what I believe is a reasonable proof-of-concept here: [nchowning/mod-webui](https://github.com/nchowning/mod-webui/commit/5f3426300388d05d5d337185a665304f124e24f1)

Feedback is welcome! I will be adding a check to make sure the user is an admin & then I'll be looking into disabling notifications for a specific period of time as @bittrance suggested.

**Comment by mohierf** [comment]
Thanks. I will have a look in the next few days ...

**Comment by mohierf** [comment]
Included in version 2.3
